library/src/types/infer.ts

```diff
@@ -14,11 +15,11 @@ export type InferInput<
   TItem extends
     | BaseSchema<unknown, unknown, BaseIssue<unknown>>
     | BaseSchemaAsync<unknown, unknown, BaseIssue<unknown>>
-    | BaseValidation<unknown, unknown, BaseIssue<unknown>>
-    | BaseValidationAsync<unknown, unknown, BaseIssue<unknown>>
-    | BaseTransformation<unknown, unknown, BaseIssue<unknown>>
-    | BaseTransformationAsync<unknown, unknown, BaseIssue<unknown>>
-    | BaseMetadata<unknown>,
+    | BaseValidation<any, unknown, BaseIssue<unknown>>
+    | BaseValidationAsync<any, unknown, BaseIssue<unknown>>
+    | BaseTransformation<any, unknown, BaseIssue<unknown>>
+    | BaseTransformationAsync<any, unknown, BaseIssue<unknown>>
+    | BaseMetadata<any>,
```

Comment:
I think `unknown` can work anywhere `BaseMetadata` is used in this file. Why do you think we should make this change?

Reply:
It is true, it works because `BaseMetadata` does not have a `._run` property. I decided to use `any` to make it consistent across all actions.